recipes-bsp/u-boot/u-boot_%.bbappend

```diff
@@ -6,7 +6,7 @@ SRC_URI:append:rpi = " \
 
 SRC_URI:append:rpi = " file://0001-rpi-always-set-fdt_addr-with-firmware-provided-FDT-address.patch"
 
-DEPENDS:append:rpi = " u-boot-default-script"
+DEPENDS:append:rpi = " ${@oe.utils.conditional('UBOOT_ENV', '', 'u-boot-default-script', '', d)}"
```

Comment:
I think this is a good idea. There are some things that I would refine: I think the core also checks for UBOOT_ENV_SUFFIX (or similar).

Reply:
OK @agherzan, I improved the commit message. From grepping the core code it seems most code that decides whether to install something or not is really hooked on UBOOT_ENV ... also UBOOT_ENV_SUFFIX has a default so it would not be an empty check. I suggest we land this as is and see if someone else runs into issues.